server/TracyProtocolServer.cpp

```diff
@@ -0,0 +1,107 @@
+    if (broadcastVersion > tracy::BroadcastVersion)
+    {
+        std::cout << "Received broadcast message with unsupported version: " << broadcastVersion << std::endl;
+        return std::nullopt;
+    }
+    switch (broadcastVersion)
+    {
+        case 3:
+        {
+            if (msgLen > sizeof(tracy::BroadcastMessage))
+            {
+                std::cout << "Received unexpected size broadcast v3 message" << std::endl;
+                return std::nullopt;
+            }
+            tracy::BroadcastMessage bm;
+            memcpy(&bm, msg, msgLen);
+            return bm;
+            break;
+        }
+        case 2:
+        {
+            if (msgLen > sizeof(tracy::BroadcastMessage_v2))
+            {
+                std::cout << "Received unexpected size broadcast v2 message" << std::endl;
+                return std::nullopt;
+            }
+            tracy::BroadcastMessage_v2 bm;
+            memcpy(&bm, msg, msgLen);
+
+            tracy::BroadcastMessage out;
+            out.broadcastVersion = broadcastVersion;
+            out.protocolVersion  = bm.protocolVersion;
+            out.activeTime       = bm.activeTime;
+            out.listenPort       = bm.listenPort;
+            strcpy(out.programName, bm.programName);
+            out.pid = 0;
+            return out;
+            break;
+        }
+        case 1:
+        {
+            if (msgLen > sizeof(tracy::BroadcastMessage_v1))
+            {
+                std::cout << "Received unexpected size broadcast v1 message" << std::endl;
+                return std::nullopt;
+            }
+            tracy::BroadcastMessage_v1 bm;
+            memcpy(&bm, msg, msgLen);
+
+            tracy::BroadcastMessage out;
+            out.broadcastVersion = broadcastVersion;
+            out.protocolVersion  = bm.protocolVersion;
+            out.activeTime       = bm.activeTime;
+            out.listenPort       = bm.listenPort;
+            strcpy(out.programName, bm.programName);
+            out.pid = 0;
+            return out;
+            break;
+        }
+        case 0:
+        {
+            if (msgLen > sizeof(tracy::BroadcastMessage_v0))
+            {
+                std::cout << "Received unexpected size broadcast v0 message" << std::endl;
+                return std::nullopt;
+            }
+            tracy::BroadcastMessage_v0 bm;
+            memcpy(&bm, msg, msgLen);
+
+            tracy::BroadcastMessage out;
+            out.broadcastVersion = broadcastVersion;
+            out.protocolVersion  = bm.protocolVersion;
+            out.activeTime       = bm.activeTime;
+            out.listenPort       = tracy::DEFAULT_CLIENT_DATA_TCP_PORT;
+            strcpy(out.programName, bm.programName);
+            out.pid = 0;
+            return out;
+            break;
+        }
+        default:
+            assert(false);
+            break;
+    }
+    return std::nullopt;
+}
+
+uint64_t ClientUniqueID(tracy::IpAddress const& addr, uint16_t port)
```

Comment:
Why is the id "unique"?

Reply:
I assume the client cannot connect with different IP addresses and ports along its lifetime, so this info can uniquely identify him ? Multicapture uses this as a key to the map of clients, so knowing if it's unique was important. 

that said, `ClientID` works too - I can change that if you prefer